generators/app/index.js

```diff
@@ -47,68 +151,48 @@ module.exports = class extends Generator {
         message:
           "What's the name of your application? This will be the default page title. Examples: '21P-530 Burials benefits form' or 'GI Bill School Feedback Tool'",
         default: 'A New Form',
+        when: !this.props.appName
       },
       {
         type: 'input',
         name: 'folderName',
         message:
           "What folder in `src/applications/` should your app live in? This can be a subfolder. Examples: 'burials' or 'edu-benefits/0993'",
-        validate: folder => {
-          if (!folder.includes(' ')) {
-            return true;
-          }
-
-          return 'Folder names should not include spaces';
-        },
-        // Remove leading and trailing forward slashes
-        filter: val => {
-          if (val.startsWith('/')) {
-            val = val.substring(1);
-          }
-          if (val.endsWith('/')) {
-            val = val.substring(0, -1);
-          }
-          return val;
-        },
+        validate: this._isInvalidFolderName,
+        filter: this._folderNameFilter,
         default: 'new-form',
+        when: !this.props.folderName
       },
       {
         type: 'input',
         name: 'entryName',
         message:
           "What should be the name of your app's entry bundle? Examples: '0993-edu-benefits' or 'feedback-tool'",
-        validate: name => {
-          if (!name.includes(' ')) {
-            return true;
-          }
-
-          return 'Bundle names should not include spaces';
+        validate: this._isInvalidEntryName,
+        default: answers => {
+          const folder = this.props.folderName ?? answers.folderName;
+          return folder.split('/').pop()
         },
-        default: answers => answers.folderName.split('/').pop(),
+        when: !this.props.entryName
       },
       {
         type: 'input',
         name: 'rootUrl',
         message:
           "What's the root url for this app? Examples: '/gi-bill-comparison-tool' or '/education/opt-out-information-sharing/opt-out-form-0993'",
-        filter: val => {
-          // Add leading slash if needed
-          if (!val.startsWith('/')) {
-            val = `/${val}`;
-          }
-          // Add `index` if a page name was not included
-          if (val.endsWith('/')) {
-            val = `${val}index`;
-          }
-          return val;
+        filter: this._rootUrlFilter,
+        default: answers => {
+          const folder = this.props.folderName ?? answers.folderName;
+          return `/${folder}`
         },
-        default: answers => `/${answers.folderName}`,
+        when: !this.props.rootUrl
       },
       {
         type: 'confirm',
         name: 'isForm',
         message: 'Is this a form app?',
         default: false,
+        when: this.props.isForm === undefined 
```

Comment:
It looks like this addition is preventing this question from being asked in the console. I tested locally, and am not asked if the app is a form app

Reply:
Yeah, this has been updated thanks.